website/docs/d/connect_instance_storage_config.html.markdown

```diff
@@ -0,0 +1,81 @@
+  instance_id    = "aaaaaaaa-bbbb-cccc-dddd-111111111111"
+  resource_type  = "CONTACT_TRACE_RECORDS"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `association_id` - (Required) The existing association identifier that uniquely identifies the resource type and storage config for the given instance ID.
+* `instance_id` - (Required) Reference to the hosting Amazon Connect Instance
+* `resource_type` - (Required) A valid resource type. Valid Values: `CHAT_TRANSCRIPTS` | `CALL_RECORDINGS` | `SCHEDULED_REPORTS` | `MEDIA_STREAMS` | `CONTACT_TRACE_RECORDS` | `AGENT_EVENTS` | `REAL_TIME_CONTACT_ANALYSIS_SEGMENTS`.
+
+## Attributes Reference
+
+In addition to all of the arguments above, the following attributes are exported:
+
+* `id` - The identifier of the hosting Amazon Connect Instance, `association_id`, and `resource_type` separated by a colon (`:`).
+* `storage_config` - Specifies the storage configuration options for the Connect Instance. [Documented below](#storage_config).
+
+### `storage_config`
+
+The `storage_config` configuration block supports the following arguments:
+
+* `kinesis_firehose_config` - A block that specifies the configuration of the Kinesis Firehose delivery stream. [Documented below](#kinesis_firehose_config).
+* `kinesis_stream_config` - A block that specifies the configuration of the Kinesis data stream. [Documented below](#kinesis_stream_config).
+* `kinesis_video_stream_config` - A block that specifies the configuration of the Kinesis video stream. [Documented below](#kinesis_video_stream_config).
+* `s3_config` - A block that specifies the configuration of S3 Bucket. [Documented below](#s3_config).
+* `storage_type` - A valid storage type. Valid Values: `S3` | `KINESIS_VIDEO_STREAM` | `KINESIS_STREAM` | `KINESIS_FIREHOSE`.
+
+#### `kinesis_firehose_config`
+
+The `kinesis_firehose_config` configuration block supports the following arguments:
+
+* `firehose_arn` - The Amazon Resource Name (ARN) of the delivery stream.
+
+#### `kinesis_stream_config`
+
+The `kinesis_stream_config` configuration block supports the following arguments:
+
+* `stream_arn` - The Amazon Resource Name (ARN) of the data stream.
+
+#### `kinesis_video_stream_config`
+
+The `kinesis_video_stream_config` configuration block supports the following arguments:
+
+* `encryption_config` - The encryption configuration. [Documented below](#encryption_config).
+* `prefix` - The prefix of the video stream. Minimum length of `1`. Maximum length of `128`. When read from the state, the value returned is `<prefix>-connect-<connect_instance_alias>-contact-` since the API appends additional details to the `prefix`.
+* `retention_period_hours` - The number of hours data is retained in the stream. Kinesis Video Streams retains the data in a data store that is associated with the stream. Minimum value of `0`. Maximum value of `87600`. A value of `0`, indicates that the stream does not persist data.
```

Comment:
Since these values are computed, do you need to specify minimum and maximum values for `prefix` and `retention_period_hours`?

Reply:
The values were used here to help practitioners understand the meaning of the number `0`. Since the minimum value was specified in this description, the maximum was included for completeness.